src/main/scala/org/stormenroute/mecha/package.scala

```diff
@@ -432,4 +432,109 @@ package object mecha {
     def default(v: =>T): Input.Query[T] = Input.default(query)(v)
   }
 
+  private[mecha] object GitIgnore {
+    import java.nio.charset.StandardCharsets._
+    import java.nio.file.{FileSystems, Paths, Files, Path}
+    import java.nio.file.StandardOpenOption._
+    import scala.collection.JavaConversions._
+
+    val log = MechaLog.Println
+
+    def readFile(file: Path): Seq[String] = {
+      Files.readAllLines(file, UTF_8)
+    }
+
+    def writeFile(file: Path, content: Seq[String]) = {
+      Files.write(file, content, UTF_8, WRITE, CREATE)
+    }
+
+    def ignore(toIgnore: String, gitIgnoreFilePath: Path, gitExcludeFilePath: Path) = {
+      val gitIgnore = if (Files.exists(gitIgnoreFilePath)) readFile(gitIgnoreFilePath) else Seq()
+      val gitExclude = if (Files.exists(gitExcludeFilePath)) readFile(gitExcludeFilePath) else Seq()
+
+      addIgnore(toIgnore, gitExclude, gitIgnore ++ gitExclude).fold(
+          {
+            case wl: Seq[WhitelistPattern] =>
+              log.warn(s"The new repo can not be ignored by Git. It is whitelisted by the following patterns: $wl")
```

Comment:
I think a WARN log should be enough so the repo is still cloned, but user can still see why it's not ignored and can manually fix it, however, I'm happy to make it stricter and fail the process with an ERROR log.